lib/src/coap_client.dart

```diff
@@ -88,6 +88,9 @@ class CoapClient {
   CoapIEndPoint? _endpoint;
   final _lock = sync.Lock();
 
+  /// Raw Public Keys for CoAPS with tinyDtls.
+  EcdsaKeys? ecdsaKeys;
```

Comment:
Probably good if `EcdsaKeys` and `PskCredentials` are implemented as our own types, such as `CoapEcdsaKeys` and `CoapPskCredentials` if it's feasible? That way we can switch the DTLS backend without affecting the user-facing API in the future, if at some point there is a native DTLS library (or SDK support).

Reply:
Sounds good, I will update the PR accordingly :)

Reply:
I reimplemented `EcdsaKeys` and `PskCredentials` (as well as a new `PskCredentialsCallback`) now within this library. The contents of those classes are then mapped to the tinydtls specific definitions. (I changed the API in the tinydtls wrapper to allow for handling PSK identity hints, which hasn't been possible before.)